diff --git a/addr.c b/addr.c
index af77f8c..de72cd4 100644
--- a/addr.c
+++ b/addr.c
@@ -80,6 +80,7 @@ say(char *line)
 
 }
 
+int
 main()
 {
     say("orc");
diff --git a/arpatok.c b/arpatok.c
index d277343..6eee48d 100644
--- a/arpatok.c
+++ b/arpatok.c
@@ -81,6 +81,7 @@ show(char *address)
 	printf(" => %s\n", q);
 }
 
+int
 main()
 {
     show("orc@pell");
diff --git a/dbm.c b/dbm.c
index cd4595e..74ca1d3 100644
--- a/dbm.c
+++ b/dbm.c
@@ -221,7 +221,7 @@ struct cmd {
 void
 usage(char *cmd, int exitcode)
 {
-    register i;
+    register int i;
     struct cmd *p;
     char *pfx = "", *opt = "";
     int pfxlen;
@@ -255,10 +255,10 @@ dbif_perror(char *text)
 }
 
 
-float
+int
 main(int argc, char **argv)
 {
-    register opt;
+    register int opt;
     struct cmd *p;
     int status;
 
diff --git a/expire.c b/expire.c
index 6bcd0db..a0b372c 100644
--- a/expire.c
+++ b/expire.c
@@ -190,7 +190,7 @@ approve(DBhandle db, char *key, int blacklist)
 }
 
 
-double
+int
 main(int argc, char **argv)
 {
     DBhandle db;
diff --git a/getif.c b/getif.c
index f4eb327..5334cc2 100644
--- a/getif.c
+++ b/getif.c
@@ -124,6 +124,7 @@ local_if_list()
 
 
 #ifdef DEBUG
+int
 main()
 {
     struct in_addr *ip = local_if_list();
diff --git a/getopt.c b/getopt.c
index 021791d..09e8805 100644
--- a/getopt.c
+++ b/getopt.c
@@ -128,7 +128,7 @@ quote(char *s, int mustquote)
 	puts(s);
 }
 
-long
+int
 main(int argc, char **argv)
 {
     int c;
diff --git a/mail.c b/mail.c
index 241d54d..93cad1c 100644
--- a/mail.c
+++ b/mail.c
@@ -43,7 +43,7 @@ static void
 mailbugcheck(struct letter *let)
 {
     int code = virus_scan(let);
-    register c;
+    register int c;
 
     if (code != 0) {
 	if (let->log != stderr) {
@@ -88,7 +88,7 @@ isto(char *line)
 static int
 collect(struct letter *let)
 {
-    register c;
+    register int c;
     enum { F, B } state;
     char *p, *q;
     int rc;
diff --git a/mbox.c b/mbox.c
index 132108a..e00c107 100644
--- a/mbox.c
+++ b/mbox.c
@@ -355,6 +355,7 @@ close_sessions()
 #ifdef DEBUG_MBOX
 struct in_addr localhost = { 0x0100007f };
 
+int
 main(int argc, char **argv)
 {
     MBOX *foo;
diff --git a/mf.c b/mf.c
index 31a6d6f..5716f16 100644
--- a/mf.c
+++ b/mf.c
@@ -615,11 +615,11 @@ void message(FILE* q,int r,char* s, ...)
 {
 }
 
-double
+int
 main(int argc, char **argv)
 {
     int i;
-    register c;
+    register int c;
     FILE *message;
     int input;
     struct letter let;
diff --git a/newaliases.c b/newaliases.c
index 5d194fb..5001474 100644
--- a/newaliases.c
+++ b/newaliases.c
@@ -190,6 +190,7 @@ newaliases(int argc, char **argv)
 
 #ifdef DEBUG
 
+int
 main()
 {
     newaliases(0);
diff --git a/ns.c b/ns.c
index e5ff2ff..97e45ce 100644
--- a/ns.c
+++ b/ns.c
@@ -195,6 +195,7 @@ struct query {
 #define NRQ	(sizeof query/sizeof query[0])
 
 
+int
 main(int argc, char **argv)
 {
     char bfr[1024];
diff --git a/postoffice.c b/postoffice.c
index 20c391c..173cfc4 100644
--- a/postoffice.c
+++ b/postoffice.c
@@ -46,7 +46,7 @@ superpowers()
 
 char *pgm;
 
-float
+int
 main(int argc, char **argv)
 {
     int opt;
diff --git a/remote.c b/remote.c
index b37a1a0..5a245e4 100644
--- a/remote.c
+++ b/remote.c
@@ -24,8 +24,8 @@ statii[10] = { PENDING, PENDING, ACCEPTED, PENDING, PENDING, REFUSED };
 static int
 SMTPwrite(MBOX *f, volatile char * volatile text, volatile unsigned long size)
 {
-    register c = 0;
-    register volatile x = 0;
+    register int c = 0;
+    register volatile int x = 0;
     void (*oldalarm)(int);
     int ret;
 
diff --git a/smtp.c b/smtp.c
index 4918373..281acd2 100644
--- a/smtp.c
+++ b/smtp.c
@@ -386,9 +386,9 @@ static int
 data(struct letter *let)
 {
 #define CRLF	0x100
-    register c = 0;
-    register c1 = CRLF;
-    register c2 = 0;
+    register int c = 0;
+    register int c1 = CRLF;
+    register int c2 = 0;
 
     if (mkspool(let) == 0) {
 	message(let->out, 452,
